exporter/elasticsearchexporter/integrationtest/datareceiver.go

```diff
@@ -42,19 +46,26 @@ const (
 	TestTracesIndex = "traces-test-idx"
 )
 
+type counters struct {
+	observedBulkRequests atomic.Int64
+}
```

Comment:
_Non-blocking nit_: I would prefer an interface that allows overriding the bulk request counter rather than this similar to the [data-provider](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/4623e7ef6c0e818db847f77d917af98a15becf18/testbed/testbed/data_providers.go#L271-L273).

So, something like `func (es *esDataReceiver) SetBulkRequestCounter(c *atomic.Uint64) { es.bulkRequestCounter = c }` with a default value set in `newElasticsearchDataReceiver`. The current interfaces kinda encourages adding more counters which I am a bit hesitant to do. Will leave it entirely up to you.

Reply:
The data provider has `SetLoadGeneratorCounters` to satisfy the `DataProvider` interface, while there isn't a ElasticsearchDataReceiver interface to satisfy. There is only 1 concrete implementation and no interface in our case. Given I can't think of any more counters to add at the moment, the extra code of `SetBulkRequestCounter` isn't offering any obvious benefits imo. We can revisit this when we find the need to make changes to esDataReceiver again.

Reply:
> The data provider has SetLoadGeneratorCounters to satisfy the DataProvider interface, while there isn't a ElasticsearchDataReceiver interface to satisfy. There is only 1 concrete implementation and no interface in our case

That was just an example and I am not suggesting introducing/satisfying any interface here. Introducing a method like `SetBulkRequestsCounter` keeps the counters required by users in their code and prevents the counters struct leaking/sharing counters. IMO, the current implementation encourages setting more counters which I am not a fan of but no strong feelings about it.

> We can revisit this when we find the need to make changes to esDataReceiver again.

SGTM!